plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,386 @@
+	fieldName := "value_" + em.Type
+	if node.FieldName != "" {
+		// set user defined field name of node
+		fieldName = node.FieldName
+	}
+
+	if sub.OutputJSONString {
+		b, err := json.Marshal(em.Value)
+		if err != nil {
+			return nil, err
+		}
+		fields := map[string]interface{}{fieldName: string(b)}
+		m := metric.New(measurement, tags, fields, t)
+		return m, nil
+	}
+
+	switch em.Type {
+	case "object":
+		flattener := jsonParser.JSONFlattener{}
+		err := flattener.FlattenJSON(fieldName, em.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		m := metric.New(measurement, tags, flattener.Fields, t)
+		return m, nil
+	case "arbool8",
+		"arint8", "aruint8",
+		"arint16", "aruint16",
+		"arint32", "aruint32",
+		"arint64", "aruint64",
+		"arfloat", "ardouble",
+		"arstring",
+		"artimestamp":
+		fields := make(map[string]interface{})
+		values := em.Value.([]interface{})
+		for i := 0; i < len(values); i++ {
+			index := strconv.Itoa(i)
+			key := fieldName + "_" + index
+			fields[key] = values[i]
+		}
+		m := metric.New(measurement, tags, fields, t)
+		return m, nil
+	case "bool8",
+		"int8", "uint8",
+		"int16", "uint16",
+		"int32", "uint32",
+		"int64", "uint64",
+		"float", "double",
+		"string",
+		"timestamp":
+		fields := map[string]interface{}{fieldName: em.Value}
+		m := metric.New(measurement, tags, fields, t)
+		return m, nil
+	}
+
+	return nil, fmt.Errorf("unsupported value type: %s", em.Type)
+}
+
+// Init is for setup, and validating config
+func (s *CtrlXDataLayerSSE) Init() error {
+	// Check all configured subscriptions for valid settings
+	for i := range s.Subscription {
+		sub := &s.Subscription[i]
+		sub.applyDefaultSettings()
+		err := choice.Check(sub.QueueBehaviour, queueBehaviours)
+		if err != nil {
+			s.Log.Infof("The right queue behaviour values are %v", queueBehaviours)
+			return fmt.Errorf("subscription %d: setting 'queue_behaviour' %q is invalid", i, sub.QueueBehaviour)
+		}
+		err = choice.Check(sub.ValueChange, valueChanges)
+		if err != nil {
+			s.Log.Infof("The right value change values are %v", valueChanges)
+			return fmt.Errorf("subscription %d: setting 'value_change' %q is invalid", i, sub.ValueChange)
+		}
+		if len(sub.Nodes) == 0 {
+			s.Log.Error("A configured subscription has no nodes configured")
+		}
+		sub.index = i
+	}
+
+	// Generate valid communication url based on configured hostname
+	u := url.URL{
+		Scheme: "https",
+		Host:   s.Hostname,
+	}
+	s.url = u.String()
+	_, err := url.Parse(s.url)
+	if err != nil {
+		return errors.New("invalid hostname")
+	}
+
+	return nil
+}
+
+// Start input as service, retain the accumulator, establish the connection.
+func (s *CtrlXDataLayerSSE) Start(acc telegraf.Accumulator) error {
+	var ctx context.Context
+	ctx, s.cancel = context.WithCancel(context.Background())
+
+	h := &httpconfig.HTTPClientConfig{
+		Timeout: s.Timeout,
+		ClientConfig: tls.ClientConfig{
+			InsecureSkipVerify: s.InsecureSkipVerify,
+		},
```

Comment:
Would it make sense to embed the `HTTPClientConfig` into your `CtrlXDataLayerSSE` struct so we also get the TLS options, proxy capabilities etc? An example can be found [in the `http` input plugin](https://github.com/influxdata/telegraf/blob/07179c466b1f22793e2d66dc68ab465d67b86fe1/plugins/inputs/http/http.go#L45).